espnet2/enh/funcs/dnn_beamformer.py

```diff
@@ -0,0 +1,298 @@
+from distutils.version import LooseVersion
+from typing import Tuple
+
+import logging
+import torch
+from torch.nn import functional as F
+from torch_complex import functional as FC
+
+from espnet.nets.pytorch_backend.frontends.beamformer import apply_beamforming_vector
+from espnet.nets.pytorch_backend.frontends.beamformer import get_mvdr_vector
+from espnet.nets.pytorch_backend.frontends.beamformer import (
+    get_power_spectral_density_matrix,  # noqa: H301
+)
+from espnet.nets.pytorch_backend.frontends.mask_estimator import MaskEstimator
+from espnet2.enh.funcs.conv_beamformer import get_covariances
+from espnet2.enh.funcs.conv_beamformer import get_WPD_filter_v2
+from espnet2.enh.funcs.conv_beamformer import perform_WPD_filtering
+from torch_complex.tensor import ComplexTensor
+
+is_torch_1_2_plus = LooseVersion(torch.__version__) >= LooseVersion("1.2.0")
+is_torch_1_3_plus = LooseVersion(torch.__version__) >= LooseVersion("1.3.0")
+
+
+class DNN_Beamformer(torch.nn.Module):
```

Comment:
Sorry, this is also what I said, but funcs is strange if it also includes classes.

`enh/layers` or `enh/modules`? `enh/layers` is better because I used the word.

Reply:
OK, I'll rename it to 'enh/layers'.

Reply:
Done.